ui/app-start-intent/src/main/java/de/danoeh/antennapod/ui/appstartintent/PlaybackSpeedActivityStarter.java

```diff
@@ -0,0 +1,40 @@
+package de.danoeh.antennapod.ui.appstartintent;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+/**
+ * Launches the playback speed dialog activity of the app with specific arguments.
+ * Does not require a dependency on the actual implementation of the activity.
+ */
+public class PlaybackSpeedActivityStarter {
+    public static final String INTENT = "de.danoeh.antennapod.intents.PLAYBACK_SPEED";
+    private final Intent intent;
+    private final Context context;
+
+    public PlaybackSpeedActivityStarter(Context context) {
+        this.context = context;
+        intent = new Intent(INTENT);
+        intent.setPackage(context.getPackageName());
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT);
+        } else {
+            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
```

Comment:
This was needed in API 19, because when clicking the playback speed button in widget it was opening the playback dialog with the screens in backstack. So I had to clear the backstack to open the dialog in "home screen" showing only the translucent activity. The only drawback is it closes the app in API 19.